src/Concerns/HasModifiers.php

```diff
@@ -52,14 +40,9 @@ public function addModifier($type, $modifier, ...$arguments)
 
     /**
      * Create a usable modifier instance
-     *
-     * @param string $type
-     * @param mixed $modifier
-     * @param array $arguments
-     * @return \Whitecube\Price\PriceAmendable
      * @throws \InvalidArgumentException
      */
-    protected function makeModifier($type, $modifier, $arguments = [])
+    protected function makeModifier(string $type, mixed $modifier, array $arguments = []): PriceAmendable
```

Comment:
`$modifier` can either be `string`, `callable`, `AbstractMoney `, `Price` or `PriceAmendable` 

```suggestion
    protected function makeModifier(string $type, string|callable|AbstractMoney|Price|PriceAmendable $modifier, array $arguments = []): PriceAmendable
```